server/protocols/webadmin/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java

```diff
@@ -96,11 +97,31 @@ public boolean testMailboxExists(String username, String mailboxName) throws Mai
             mailboxSession);
     }
 
-    public void deleteMailbox(String username, String mailboxName) throws MailboxException, UsersRepositoryException {
+    public void deleteMailbox(String username, String mailboxName) throws MailboxException, UsersRepositoryException, MailboxHaveChildrenException {
         usernamePreconditions(username);
         Preconditions.checkArgument(!Strings.isNullOrEmpty(mailboxName));
         MailboxSession mailboxSession = mailboxManager.createSystemSession(USER_NAME, LOGGER);
-        deleteMailbox(mailboxSession, convertToMailboxPath(username, mailboxName, mailboxSession));
+        MailboxPath mailboxPath = convertToMailboxPath(username, mailboxName, mailboxSession);
+        try {
+            if (!haveChildren(mailboxPath, mailboxSession)) {
+                deleteMailbox(mailboxSession, mailboxPath);
+            } else {
+                throw new MailboxHaveChildrenException(mailboxName);
+            }
+        } catch (MailboxNotFoundException e) {
+            LOGGER.info("Attempt to delete mailbox {} for user {} that does not exists", mailboxPath.getName(), mailboxPath.getUser());
+        }
+    }
+
+    private boolean haveChildren(MailboxPath mailboxPath, MailboxSession mailboxSession) throws MailboxException {
+        return mailboxManager.search(
+            MailboxQuery.builder()
+                .base(mailboxPath)
+                .build(), mailboxSession)
+            .stream()
+            .findAny()
```

Comment:
With filter, how you differenciate no result at all from (searching for a mailbox that do not exist) from searching a mailbox without child ?